### ISSUE TITLE
LTO triggers format-truncation warning in econftool (-Werror)

### ISSUE THREAD

**Issue by andhe** [openSUSE/libeconf#167]
Original downstream bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1015485

Apparently enabling LTO[1] implicitly enables `-Wformat-truncation` and since econftool.c uses sprintf to truncate strings this triggers the warning, which in turn becomes a build error[2] since `-Werror` is also added to cflags in `meson.build`.

This can also be tested/triggered by adding `-Werror=format-truncation` to CFLAGS.

As a side note always enabling Werror is a problems for distributions as it can introduce lots of build failures on compiler upgrades/changes. Meson even warns about this and suggests instead using the builtin option (-Dwerror=true).
```
../meson.build:42: WARNING: Consider using the built-in werror option instead of using "-Werror".
```
Dropping `-Werror` from meson.build also "fixes" this (format truncation remains ofcourse, but warnings no longer being treated as errors means there's no build failure).

[1]:
```
--- libeconf-0.4.4+dfsg1.orig/meson.build
+++ libeconf-0.4.4+dfsg1/meson.build
@@ -20,6 +20,8 @@ add_project_arguments(['-D_GNU_SOURCE=1'

 possible_cc_flags = [
                  '-Werror',
+                  '-flto=auto',
+                  '-ffat-lto-objects',
                  '-fstack-protector-strong',
                  '-funwind-tables',
                  '-fasynchronous-unwind-tables',
```

[2]: see http://qa-logs.debian.net/2022/06/09/dpkglto/libeconf_0.4.4+dfsg1-1_unstable_dpkglto.log

**Comment by schubi2** [comment]
Hallo, I like your options, have adapted the code and have released 0.4.7.
Thanks for the hints !

**Comment by schubi2** [comment]
closing....
